### ISSUE TITLE
[BUG] Spring Boot app getting crash at startup due to JobRunr cron job 'JobRunrException'

### ISSUE THREAD

**Issue by RahulSDeshpande** [jobrunr/jobrunr#189]
**Are you running the latest version of JobRunr?**
Yes

**How to Reproduce**
I am strangely facing similar issue since yesterday.

I don't know what is wrong going, I didn't change any line of code in this file since long.

Here is my code:
```kt
@PostConstruct
fun startJobRunr() {
    jobScheduler.scheduleRecurrently(Cron.minutely()) {
        runJob1()
        runJob2()
    }
}

@Job
fun runJob2() {
    ..
}

@Job
fun runJob2() {
    ..
}
```

**Crash log stack-trace**
```kt
Caused by: org.jobrunr.JobRunrException: JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace)
	at org.jobrunr.JobRunrException.shouldNotHappenException(JobRunrException.java:43) ~[jobrunr-3.2.1.jar:na]
	at org.jobrunr.jobs.details.JobDetailsGeneratorUtils.createObjectViaMethod(JobDetailsGeneratorUtils.java:70) ~[jobrunr-3.2.1.jar:na]
	at org.jobrunr.jobs.details.instructions.JobDetailsInstruction.getObject(JobDetailsInstruction.java:79) ~[jobrunr-3.2.1.jar:na]
	at org.jobrunr.jobs.details.instructions.JobDetailsInstruction.invokeInstruction(JobDetailsInstruction.java:44) ~[jobrunr-3.2.1.jar:na]
	at org.jobrunr.jobs.details.instructions.AbstractJVMInstruction.invokeInstructionAndPushOnStack(AbstractJVMInstruction.java:18) ~[jobrunr-3.2.1.jar:na]
	at org.jobrunr.jobs.details.JobDetailsBuilder.invokeInstructions(JobDetailsBuilder.java:92) ~[jobrunr-3.2.1.jar:na]
	at org.jobrunr.jobs.details.JobDetailsBuilder.getJobDetails(JobDetailsBuilder.java:70) ~[jobrunr-3.2.1.jar:na]
	at org.jobrunr.jobs.details.AbstractJobDetailsFinder.getJobDetails(AbstractJobDetailsFinder.java:84) ~[jobrunr-3.2.1.jar:na]
	at org.jobrunr.jobs.details.KotlinJobDetailsFinder.getJobDetails(KotlinJobDetailsFinder.java:16) ~[jobrunr-3.2.1.jar:na]
	at org.jobrunr.jobs.details.JobDetailsAsmGenerator.toJobDetails(JobDetailsAsmGenerator.java:16) ~[jobrunr-3.2.1.jar:na]
	at org.jobrunr.scheduling.JobScheduler.scheduleRecurrently(JobScheduler.java:542) ~[jobrunr-3.2.1.jar:na]
	at org.jobrunr.scheduling.JobScheduler.scheduleRecurrently(JobScheduler.java:506) ~[jobrunr-3.2.1.jar:na]
	at org.jobrunr.scheduling.JobScheduler.scheduleRecurrently(JobScheduler.java:472) ~[jobrunr-3.2.1.jar:na]
	at ******* HIDDEN AS ITS A PRIVATE PROJECT FILE PATH **.startJobRunr(ScheduledTasks.kt:97) ~[classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_282]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_282]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_282]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_282]
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleElement.invoke(InitDestroyAnnotationBeanPostProcessor.java:389) ~[spring-beans-5.3.5.jar:5.3.5]
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:333) ~[spring-beans-5.3.5.jar:5.3.5]
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:157) ~[spring-beans-5.3.5.jar:5.3.5]
```

**Environment**
I'm using JobRunr version: `3.2.1`

**My IntelliJ config**
```kt
IntelliJ IDEA 2021.1.1 (Ultimate Edition)
Build #IU-211.7142.45, built on April 30, 2021
Runtime version: 11.0.10+9-b1341.41 x86_64
VM: Dynamic Code Evolution 64-Bit Server VM by JetBrains s.r.o.
macOS 11.2.3
GC: ParNew, ConcurrentMarkSweep
Memory: 4012M
Cores: 12
Non-Bundled Plugins: com.markskelton.one-dark-theme (5.3.0), net.ashald.envfile (3.2.1), wu.seal.tool.jsontokotlin (3.7.2), org.jetbrains.plugins.vue (211.7142.13), org.jetbrains.kotlin (211-1.5.21-release-317-IJ7442.40), com.nbadal.ktlint (0.7.6), ru.adelf.idea.dotenv (2021.1)
Kotlin: 211-1.5.21-release-317-IJ7442.40
```

**Expected behavior**
All the jobs should run & the spring boot instance should not crash.

**Comment by rdehuyss** [comment]
The caused by of the stacktrace is missing so I can't really help now.

**Comment by rdehuyss** [comment]
Did your kotlin version change?

**Comment by RahulSDeshpande** [comment]
@rdehuyss 

That is the only stack trace I see in the logs.

And no I didn't change my Kotlin version.

**Comment by rdehuyss** [comment]
Yes, it's the only stack trace but the caused by part is missing. See https://github.com/jobrunr/jobrunr/blob/master/core/src/main/java/org/jobrunr/jobs/details/JobDetailsGeneratorUtils.java#L70 (the variable e)/

**Comment by RahulSDeshpande** [comment]
Hi @rdehuyss

I finally fixed it.

Seems like, a single `scheduleRecurrently() { .. }` won't accept multiple job method calls 😄
So I wrote multiple `scheduleRecurrently() { .. }`s for all the job method calls.

This is the final code.

```kt
@PostConstruct
fun startJobRunr() {

    jobScheduler.scheduleRecurrently(Cron.minutely()) {
        runJob1()
    }

    jobScheduler.scheduleRecurrently(Cron.minutely()) {
        runJob2()
    }
}

@Job
fun runJob2() {
    ..
}

@Job
fun runJob2() {
    ..
}
```

I am not sure whether this is mentioned in the JobRunr docs or I missed reading it.

But this is strange.
The library should allow multiple jobs calls from a single `scheduleRecurrently` call I think.

Is this design by purpose?

Anyways great library!

Cheers 💯

**Comment by RahulSDeshpande** [comment]
Closing this issue 👍🏼

**Comment by rdehuyss** [comment]
Documentation needs to be updated

**Comment by rdehuyss** [comment]
Solved in 4.0.0-RC2 (somewhere tomorrow in Maven Central) and in docs once deployed
